### PR TITLE
chore: [#8] Integrated Lint Staged

### PR DESCRIPTION
**Pull request by eldarcelik**
### What has changed and why?
Integrated Lint Staged. Packages that are installed:
- **`lint-staged`** helps you run various scripts (typically linters or code formatters) on files that are staged for a commit in a version control.
- **`husky`** allows you to easily set up and manage Git hooks in your project. It is configured pre-commit hook.
- **`package.json`** clean up.
- Closes: #8 

### Branch Name Validation

- [x] Branch name pattern: `{type}/{JIRA-ID}` or `{type}/issue-{GITHUB-ISSUE-ID}`

### PR Title Validation

- [x] PR title pattern: `{type}: [{JIRA-ID}] Description` or `{type}: [#{GITHUB-ISSUE-ID}] Description`

### Only check what is relevant for this PR

- [ ] Styling verified for different screen sizes and browsers
- [ ] Functionality verified for different screen sizes and browsers
- [ ] Relevant unit tests added/updated
- [ ] Storybook files added/updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,23 @@
...
-    "stylelint:fix": "stylelint --fix **/*.css"
+    "stylelint:fix": "stylelint --fix **/*.css",
+    "prepare": "husky install"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
```
Comment (4zuko) [inline package.json:22]: Could you please move the configuration to `lint-staged.config.js` that should be added to the project root? I want to have cleaner and more readable `package.json`.

File: package.json [hunk 2]
```diff
@@ -11,7 +11,23 @@
     "lint": "eslint .",
     "lint:fix": "eslint --fix .",
     "stylelint": "stylelint **/*.css",
-    "stylelint:fix": "stylelint --fix **/*.css"
+    "stylelint:fix": "stylelint --fix **/*.css",
+    "prepare": "husky install"
```
Comment (4zuko) [inline package.json:15]: Add npm script `"precommit": "lint-staged",`

Comment (4zuko) [reply package.json:15]: Have this as reminder to add it once jest is configured:

`"prepush": "jest --bail --changedSince=origin/main --coverage --maxWorkers=50%",`

File: .husky/pre-commit
Comment (4zuko) [inline .husky/pre-commit:1]: You should have pre-commit and pre-push scripts:

// pre-commit

```shell
#!/usr/bin/env sh

. "$(dirname -- "$0")/_/husky.sh"

yarn precommit ## TODO: Replace this command with npm command
```

// pre-push
```shell
#!/usr/bin/env sh

. "$(dirname -- "$0")/_/husky.sh"

npm run prepush
```

File: package.json [hunk 3]
```diff
@@ -11,7 +11,23 @@
...
+  },
+  "husky": {
```
Comment (4zuko) [inline package.json:17]: I think that this is not required but please double check it official documentation.

**Comment by 4zuko** [comment]
Just a note. I use the following configuration in my projects:

```js
// .lintstagedrc.js

module.exports = {
  '*.{js,jsx,ts,tsx,scss}': ['prettier --write'],
  '*.scss': ['stylelint --custom-syntax postcss-scss'],
  '*.{js,jsx,ts,tsx}': ['eslint'],
}
```

```js
// .lintstagedrc.ts.js

const config = require('./.lintstagedrc.base')

module.exports = {
  ...config,
  '**/*.ts?(x)': () => 'tsc -p tsconfig.json --noEmit',
}
```

File: package.json (hunk 1 above)
Comment (eldarcelik) [reply package.json]: I moved configuration to **`lint-staged.config.js`** file.

File: .husky/pre-commit
Comment (eldarcelik) [reply .husky/pre-commit:1]: I will add **`prepush`** after implementing Jest.

File: package.json (hunk 3 above)
Comment (eldarcelik) [reply package.json]: You are right, there is no need to specify this.

File: lint-staged.config.json
Comment (4zuko) [inline lint-staged.config.json:1]: @eldarcelik: Try with:

```js
// lint-staged.config.js

module.exports = {
  "*.{ts,tsx,css,json}": ["prettier --write"],
  "*.{ts,tsx,json}": ["eslint --fix"],
  "*.css": ["stylelint --fix"]
}
```

**Review by 4zuko — APPROVED** [review]
**`LGTM`**

**Pull request merged**